launcher/src/main/java/org/wildfly/core/launcher/DomainCommandBuilder.java

```diff
@@ -522,8 +523,17 @@ public DomainCommandBuilder addHostControllerJavaOption(final String arg) {
      */
     public DomainCommandBuilder addHostControllerJavaOptions(final String... args) {
         if (args != null) {
-            for (String arg : args) {
-                addHostControllerJavaOption(arg);
+            for (int i = 0; i < args.length; i++) {
+                if (Arguments.isModuleSystemArgument(args[i])) {
+                    // handle java module system options
+                    if (i < args.length - 1) {
+                        this.hostControllerJavaOpts.add(Arguments.JavaModuleSystemArgument.of(args[i], args[++i]));
+                    } else {
+                        // ignore corrupted module system argument
```

Comment:
Same as above, we should just add the arguments and allow the command to fail when executed.